### DISCUSSION TITLE
Integration with foreign languages

### DISCUSSION THREAD

**Discussion by KonstantinosPetrakis** [langroid/langroid#571] [Q&A]
Hello,

First of all, great work on the project! It’s been straightforward to follow, and I’m particularly impressed with how well it supports local LLMs compared to commercial alternatives like Langraph, which can sometimes feel overengineered and seem less focused on local models.

I have a question regarding the library’s performance in foreign languages. Have you tested it in such contexts? I’m not familiar with the internal workings, so I’m unsure if there are any language-specific limitations or if it’s designed to be used exclusively with English.

For example, if I were to use a multilingual LLM and create agent tasks with system_message in a foreign language, and define ToolMessages with the purpose attribute in that language, would the library function as expected?

If the library is primarily intended for English, I was considering creating two tools that leverage a translation model/library (such as LibreTranslate) to translate between English and the foreign language.

Tell me what you think. 

Thanks.

**Comment by KonstantinosPetrakis** [comment]
As I understand it, your framework provides additional instructions to the LLM at some point, such as details about available tools, required formats, and so on.

It seems likely that these instructions are currently provided in English—please correct me if I’m mistaken.

Have you considered translating these instructions into other languages? For instance, using a tool like gettext to handle multilingual translations could be an interesting approach.

**Reply by pchalasani** [reply]
Thanks @KonstantinosPetrakis glad you like the library! 

Yes currently it is all English based and we haven't considered non-english abilities. I am open to contributions that can extend it to other languages.
